src/Symfony/Component/DependencyInjection/ParameterBag/ParameterBag.php

```diff
@@ -288,4 +288,21 @@ public function unescapeValue($value)
 
         return $value;
     }
+
+    /**
+     * @internal to be removed in 4.0
+     */
+    public static function normalizeName($name)
+    {
+        if (0 === strpos($name, 'env(') && ')' === substr($name, -1) && 'env()' !== $name) {
+            return strtolower($name);
+        }
+
+        $normalizedName = strtolower($name);
+        if ($normalizedName !== (string) $name) {
+            @trigger_error(sprintf('Parameter names will be made case sensitive in Symfony 4.0. Using "%s" instead of "%s" is deprecated since version 3.4.', $name, $normalizedName), E_USER_DEPRECATED);
```

Comment:
So this basically means that I get a deprecation message whenever I use uppercase characters in parameter names? This is kind of misleading because uppercase letters won't be forbidden in Symfony 4, will they?

I mean, if I set a parameter like `myCamelCaseParameter` and always use `myCamelCaseParameter` to fetch it from the bag, I don't see a problem with that, although it might not be best practice to format parameter names that way.

The problematic workflow would be to set `myCamelCaseParameter` and request `mycamelcaseparameter` afterwards. Because, as soon as parameter names are treated case-sensitively, this piece of code is going to break. And imho only in this case, I should get a deprecation warning.